TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/cluster.py

```diff
@@ -51,21 +44,29 @@ def __init__(self, config: Dict) -> None:
         self.endpoint = config["endpoint"]
         if self.endpoint.startswith("https"):
             self.allow_insecure = config.get("allow_insecure", False)
-        self.auth_type = AuthMethod[config["authorization"]["type"].upper()]
-        self.auth_details = config["authorization"]["details"]
-        pass
+        if "authorization" in config:
+            self.auth_type = AuthMethod[config["authorization"]["type"].upper()]
+            self.auth_details = config["authorization"]["details"]
 
     def call_api(self, path, method: HttpMethod = HttpMethod.GET) -> requests.Response:
         """
         Calls an API on the cluster.
         """
         if self.auth_type == AuthMethod.BASIC:
-            auth = HTTPBasicAuth(self.auth_details["username"], self.auth_details["password"])
+            assert self.auth_details is not None  # for mypy's sake
+            auth = HTTPBasicAuth(
+                self.auth_details.get("username", None), self.auth_details["password"]
```

Comment:
Why use a different pattern for getting username versus getting password here?

Reply:
good question. No good reason as far as I can tell.